lib/Doctrine/DBAL/Connection.php

```diff
@@ -1589,11 +1589,17 @@ public function ping()
             return $this->_conn->ping();
         }
 
+        set_error_handler(function ($severity, $message) {
+            throw new \PDOException($message, $severity);
```

Comment:
I doubt it is the correct way to set an error handler here that throws a `\PDOException` as not all of the drivers are PDO drivers. Furthermore `Doctrine\DBAL\Connection` should only throw `Doctrine\DBAL\DBALException` based exceptions that wrap the original driver exception.
This is more of a general PDOMySQL driver issue which should be solved in the driver implementation itself (if possible).
In the end it looks like a [PHP bug](https://bugs.php.net/bug.php?id=63812) because when setting `\PDO::ERRMODE_EXCEPTION` on a PDO connection, no warnings are expected to be risen.